### PR TITLE
Refactor benchmarks

### LINKED ISSUE #561: [ENHANCEMENT]: Add `NumInputs` axis to all benchmarks with value `default::N`
### Is your feature request related to a problem? Please describe.

It can be useful to sweep the item count when studying cuco benchmarks. I recommend adding an nvbench axis for `NumInputs` to each benchmarks where the axis could make sense. We should set the axis to only run the default value, so as to not change the benchmark runtime or default behavior. Adding the axis allows us to vary each benchmark at runtime.

### Describe the solution you'd like

I believe adding this line and updating the benchmarks to use `NumInputs` would suffice.
`.add_int64_axis("NumInputs", defaults::N);`

### Describe alternatives you've considered

Edit the cuco benchmarks and rebuild

### Additional context

It might also be useful to add placeholder axes for `OCCUPANCY`, `MULTIPLICITY` and `MATCHING_RATE` that set to the default values. Perhaps like: 
```
  .add_float64_axis("Occupancy", defaults::OCCUPANCY);
  .add_float64_axis("MatchingRate", defaults::MATCHING_RATE);
```

### PR DESCRIPTION
**Pull request by sleeepyjack**
Closes #561 

This PR includes:

- Expose all relevant benchmark axes so they can be set from the CLI (see #561)
- Use `_async` variants whenever possible to get more accurate results
- Add `_capacity` benchmarks for all data structures
- Unify ordering of benchmarks
- Restructure the benchmark directory
  - remove `benchmarks/analysis/` directory
  - remove `benchmarks/hash_table/` sub-directory and flatten structure

### CODE REVIEW

File: benchmarks/static_map/insert_bench.cu [hunk 1]
```diff
@@ -94,13 +85,27 @@ NVBENCH_BENCH_TYPES(static_map_insert,
   .set_name("static_map_insert_unique_occupancy")
   .set_type_axes_names({"Key", "Value", "Distribution"})
   .set_max_noise(defaults::MAX_NOISE)
+  .add_int64_axis("NumInputs", {defaults::N})
```
Comment (PointKernel) [inline benchmarks/static_map/insert_bench.cu:88]: shall we add the `Multiplicity` axis to all test cases?

Comment (sleeepyjack) [reply benchmarks/static_map/insert_bench.cu:88]: I thought about this too. Multiplicity/Skew are parameters of the input distribution. If we select a unique key distribution, then the Multiplicity/Skew values in those extra columns are completely unrelated. I'd say that is confusing.

Comment (PointKernel) [reply benchmarks/static_map/insert_bench.cu:88]: Good point.

This test is dedicated to `unique` input and we can vary multiplicity in `static_map_insert_uniform_multiplicity`.

**Review by PointKernel — APPROVED** [review]
@sleeepyjack can you please make sure there are no more invocations of `get_*_or_default`? Otherwise, looks great to me. Thank you!

**Comment by sleeepyjack** [comment]
@PointKernel I did another pass fixing some minor stuff. This PR should be ready to go once CI agrees.

@GregoryKimball unfortunately this will likely break the current nightly benchmark scripts or at least lead to some sudden jumps in the charts. I apologize in advance.

File: benchmarks/dynamic_map/insert_bench.cu [hunk 2]
```diff
@@ -82,10 +82,12 @@ NVBENCH_BENCH_TYPES(dynamic_map_insert,
...
                                       defaults::VALUE_TYPE_RANGE,
                                       nvbench::type_list<distribution::unique>))
-  .set_name("dynamic_map_insert_unique_num_inputs")
+  .set_name("dynamic_map_insert_unique_capacity")
   .set_type_axes_names({"Key", "Value", "Distribution"})
   .set_max_noise(defaults::MAX_NOISE)
-  .add_int64_axis("NumInputs", defaults::N_RANGE);
+  .add_int64_axis("NumInputs", defaults::N_RANGE)
+  .add_int64_axis("InitSize", {defaults::INITIAL_SIZE})
+  .add_float64_axis("BatchSize", {defaults::BATCH_SIZE});
```
Comment (sleeepyjack) [inline benchmarks/dynamic_map/insert_bench.cu]:
```suggestion
  .add_int64_axis("BatchSize", {defaults::BATCH_SIZE});
```

File: benchmarks/dynamic_map/insert_bench.cu [hunk 3]
```diff
@@ -94,6 +96,9 @@ NVBENCH_BENCH_TYPES(dynamic_map_insert,
   .set_name("dynamic_map_insert_uniform_multiplicity")
   .set_type_axes_names({"Key", "Value", "Distribution"})
   .set_max_noise(defaults::MAX_NOISE)
+  .add_int64_axis("NumInputs", {defaults::N})
+  .add_int64_axis("InitSize", {defaults::INITIAL_SIZE})
+  .add_float64_axis("BatchSize", {defaults::BATCH_SIZE})
```
Comment (sleeepyjack) [inline benchmarks/dynamic_map/insert_bench.cu]:
```suggestion
  .add_int64_axis("BatchSize", {defaults::BATCH_SIZE})
```

File: benchmarks/dynamic_map/insert_bench.cu [hunk 4]
```diff
@@ -103,4 +108,7 @@ NVBENCH_BENCH_TYPES(dynamic_map_insert,
   .set_name("dynamic_map_insert_gaussian_skew")
   .set_type_axes_names({"Key", "Value", "Distribution"})
   .set_max_noise(defaults::MAX_NOISE)
+  .add_int64_axis("NumInputs", {defaults::N})
+  .add_int64_axis("InitSize", {defaults::INITIAL_SIZE})
+  .add_float64_axis("BatchSize", {defaults::BATCH_SIZE})
```
Comment (sleeepyjack) [inline benchmarks/dynamic_map/insert_bench.cu]:
```suggestion
  .add_int64_axis("BatchSize", {defaults::BATCH_SIZE})
```

**Pull request merged**